conference/src/test/java/com/kyriba/conference/management/api/HallRestControllerIntegrationTest.java

```diff
@@ -0,0 +1,289 @@
+package com.kyriba.conference.management.api;
+
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.kyriba.conference.management.domain.dto.HallRequest;
+import com.kyriba.conference.management.domain.dto.HallResponse;
+import com.kyriba.conference.management.domain.exception.EntityIsUsedException;
+import com.kyriba.conference.management.domain.exception.EntityNotFoundException;
+import com.kyriba.conference.management.domain.exception.SameEntityExistsException;
+import com.kyriba.conference.management.service.HallService;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+
+import javax.validation.ConstraintViolationException;
+import java.util.List;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.tuple;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.refEq;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.springframework.http.MediaType.APPLICATION_JSON_UTF8_VALUE;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(HallController.class)
+@ActiveProfiles("test")
+public class HallRestControllerIntegrationTest
+{
+  @Rule
+  public ExpectedException exception = ExpectedException.none();
+
+  @Autowired
+  private MockMvc mockMvc;
+
+  @Autowired
+  private ObjectMapper objectMapper;
+
+  @MockBean
+  private HallService hallService;
+
+  private HallRequest hallRequest = new HallRequest("My Hall #1", 10);
+
+
+  @Test
+  public void createHallEndpointTest() throws Exception
```

Comment:
Try to give tests logical name to describe tested scenarios like "hallIsCreatedWithUniqueId".